### PR TITLE
Decline pending group invites from user, when user is blocked

### LINKED ISSUE #12839: Decline non-answered group chat invitation when a user is blocked
# Feature Issue

leftover from https://github.com/status-im/status-react/issues/12108

As a user
When I block someone
I want existing group chat invitation to be declined
So that it does not clutter my home screen

### Description

When a user is blocked, existing group chat invitation that have not been accepted  should also be removed.

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-react/issues/12839

leftover from https://github.com/status-im/status-react/pull/12835

### Summary
PR implements a feature, which auto declines pending group invites from a user when he/she is blocked. Group invites on the home screen and also in the activity center will be declined.

#### status-go PR:
https://github.com/status-im/status-go/pull/2455

### CODE REVIEW

File: src/status_im/transport/message/core.cljs [hunk 1]
```diff
@@ -81,9 +81,7 @@
         (fx/merge cofx
                   (process-next response-js sync-handler)
                   (models.contact/ensure-contacts
-                   (map data-store.contacts/<-rpc contacts-clj)
-                   (when chats
-                     (map data-store.chats/<-rpc (types/js->clj chats))))))
+                   (map data-store.contacts/<-rpc contacts-clj) chats)))
```
Comment (flexsurfer) [inline src/status_im/transport/message/core.cljs:84]: could you elaborate, i don't see changes in ensure-contacts, how it will work ?

Comment (Parveshdhull) [reply src/status_im/transport/message/core.cljs:84]: actually ensure-contacts is just a middle man, chats are only used in `contact-blocked` function of `block.cljs`

Comment (Parveshdhull) [reply src/status_im/transport/message/core.cljs:84]: https://github.com/status-im/status-react/pull/12878/files#diff-bec107b652230a20367afcf97c75c0d66b35f99cbca6484d59a9ddd17f4422a5

Comment (flexsurfer) [reply src/status_im/transport/message/core.cljs:84]: i see

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
thank you for PR, @Parveshdhull !
Several questions:
1 ) Your branch in go is based on the last stable version? 
If not, can we please rebase it (I see strange behavior in public chats, but don't think it is related to your changes)?
2) We still have some notifications from blocked  user in activity center (screen). they will be addressed separately, yes?
IMAGE 2021-12-03 15:40:09

**Comment by Parveshdhull** [comment]
@churik, Thanks for testing the PR.
Sure, I will rebase the status-go, and see if second issue can also be addressed, simultaneously 👍

**Comment by Parveshdhull** [comment]
Done👍, hopefully, both issues should be fixed now 🤞

**Comment by churik** [comment]
@Parveshdhull 
Issue 2 is still here.
Issue 1 is related to https://github.com/status-im/status-react/issues/12884 
should I report issue 2 separately?

**Comment by Parveshdhull** [comment]
Yes, reporting separate sounds good 👍, Thanks

**Pull request merged**